mona_sdk/authentication.py

```diff
@@ -243,12 +243,16 @@ def _calculate_and_set_time_to_refresh(api_key):
         )
 
 
-def _handle_authentications_error(error_message, should_raise_exception):
+def _handle_authentications_error(
+    error_message, should_raise_exception, args_to_log=None
```

Comment:
nit - here this new param shouldn't be called args_to_log - it's passed as the message that was failed to be sent to mona, so should be renamed to failed_message or something.

Also it will be more readable if instead of passing args[1] to this function inline, you would either save it as a local variable there or explicitly say failed_message=args[1] to make it more readable that this is what the second arg is...

Reply:
I had a bit of a problem here, since this decorator also wraps upload_config, args[1] might be the new config to upload. 
I mentioned it in the 413 line comment, but its not very straightforward.. wdyt?
@nemo-mona

Reply:
you mean line 314 :-) 

that's fine. What I meant was just to store the inline term (args[1] if should_log_args else None) in a local variable with a name that will hint about what it's supposed to be, and pass that in the next line to _handle_authentication_error... Just for readabilility.

Also any reason not to rename args_to_log?

Reply:
I just don't want to rename it to failed_messages because it can also be a config to upload (and I saved the "args[1] if should_log_args else None" as a local variable)
@nemo-mona